watcher/lib/influxdb/rollups.go

```diff
@@ -119,38 +119,68 @@ func buildQuery(r Rollup) (string, error) {
 	return b.String(), nil
 }
 
+// funcsWithParams defines which functions need an additional parameter
+var funcsWithParams = []string{
+	constants.FunctionPercentile,
+	constants.FunctionBottom,
+	constants.FunctionTop,
+	constants.FunctionSample,
+}
+
+// isFuncWithParams checks if function is one of the composable Functions listed above
+func isFuncWithParams(funcName string) bool {
+	for _, name := range funcsWithParams {
+		if name == funcName {
+			return true
+		}
+	}
+	return false
+}
+
 // buildFunction returns a function string based on the provided function configuration
 func buildFunction(f Function) (string, error) {
 	alias := f.Alias
 	if alias == "" {
 		alias = f.Field
 	}
-	if strings.HasPrefix(f.Function, constants.FunctionPercentile) {
-		value, err := parsePercentileValue(f.Function)
+
+	// split function name, based on the "_" separator (eg: percentile_99, top_10, ecc)
+	funcAndValue := strings.Split(f.Function, "_")
+	if len(funcAndValue) == 2 && isFuncWithParams(funcAndValue[0]) {
```

Comment:
There's a bug here: incorrectly formatted "percentile" (or other composite func) is ignored because it will not pass the `len == 2` check and proceed straight to the final return resulting in incorrect function. For example, somebody provides just `percentile`. Previous code ensured that percentile function has 2 parts.

I think that the logic should be inverted here:

1. First, you check that the f.Function _is_ one of composite functions (e.g. by checking prefix).
  a. If it is not a composite function, you return like line 159 does (135 in old code).
  b. If it is a composite function, you check that:
    - It has two parts.
    - The value is formatted correctly (e.g. your `validateParam`).

Reply:
Fixed...  I think (?) :crab: